core/embed/lib/unit_variant.c

```diff
@@ -1,23 +1,54 @@
-#include "unit_variant.h"
+#include <string.h>
+
 #include "flash_otp.h"
 #include "model.h"
+#include "unit_variant.h"
+#include TREZOR_BOARD
 
 static uint8_t unit_variant_color = 0;
 static bool unit_variant_btconly = false;
 static bool unit_variant_ok = false;
 
+static int16_t unit_variant_build_year = -1;
+
 static void unit_variant_0x01(const uint8_t *data) {
   unit_variant_color = data[1];
   unit_variant_btconly = data[2] == 1;
   unit_variant_ok = true;
 }
 
+static int16_t unit_variant_get_build_year(void) {
+  uint8_t data[FLASH_OTP_BLOCK_SIZE] = {0};
+
+  secbool result =
+      flash_otp_read(FLASH_OTP_BLOCK_BATCH, 0, data, FLASH_OTP_BLOCK_SIZE);
+
+  if (sectrue != result || data[0] == 0xFF) {
+    return -1;
+  }
+
+  size_t len = strnlen((char *)data, FLASH_OTP_BLOCK_SIZE);
+
+  for (int i = 0; i < len; i++) {
```

Comment:
I didn't find any info about how is OTP organized except this: https://docs.trezor.io/trezor-firmware/core/misc/memory.html?highlight=otp#otp 
but it's not apparent what exact data is there byte-wise.

Can we add a comment here explaining what are we parsing? Something like: `block 0 contains 'somedata-YYMMDD'`. Thanks.

Reply:
Yes, we use `{PREFIX}-YYMMDD` where `{PREFIX}` is defined here:

https://github.com/trezor/trezor-firmware/blob/633107644487480286a7b5fe9e1dea36a1efc9e2/core/embed/prodtest/main.c#L52-L56

the control read is happening here (maybe we should update the comment on line 579?):

https://github.com/trezor/trezor-firmware/blob/633107644487480286a7b5fe9e1dea36a1efc9e2/core/embed/prodtest/main.c#L578-L585

Reply:
updated the comment and readme: https://github.com/trezor/trezor-firmware/commit/43768b56255f798baa49a026e6262a0864a54bfd

Reply:
and a comment in unit_variant: https://github.com/trezor/trezor-firmware/commit/f1af347bfe5a4e9b2bbbdf77a6f88a6590934c49